In [1]:
import numpy as np
import pandas as pd

In [2]:
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [3]:
sns.set()

In [4]:
data = pd.read_csv('iris.csv')

In [5]:
data.head()

5.1  3.5  1.4  0.2  Iris-setosa
0  4.9  3.0  1.4  0.2  Iris-setosa
1  4.7  3.2  1.3  0.2  Iris-setosa
2  4.6  3.1  1.5  0.2  Iris-setosa
3  5.0  3.6  1.4  0.2  Iris-setosa
4  5.4  3.9  1.7  0.4  Iris-setosa

In [13]:
data_groupd = data.groupby(['Iris-setosa'],as_index=False).mean()

In [14]:
data_groupd

Iris-setosa       5.1       3.5       1.4       0.2
0      Iris-setosa  5.004082  3.416327  1.465306  0.244898
1  Iris-versicolor  5.936000  2.770000  4.260000  1.326000
2   Iris-virginica  6.588000  2.974000  5.552000  2.026000

In [15]:
data.describe()

5.1         3.5         1.4         0.2
count  149.000000  149.000000  149.000000  149.000000
mean     5.848322    3.051007    3.774497    1.205369
std      0.828594    0.433499    1.759651    0.761292
min      4.300000    2.000000    1.000000    0.100000
25%      5.100000    2.800000    1.600000    0.300000
50%      5.800000    3.000000    4.400000    1.300000
75%      6.400000    3.300000    5.100000    1.800000
max      7.900000    4.400000    6.900000    2.500000

From the data above, iris-virginica seems to be the specie with the largest mean sepal width
ampng the other two iris species. Also, sepal width "5.1" is the one with the highest representation among the sepal width